cmd/main.go

```diff
@@ -277,6 +251,71 @@ func runDev(ctx context.Context, args []string) error {
 	return cmd.Wait()
 }
 
+func parseGoListJson(out []byte) (currentModule, moduleDir string, replacements []xcaddy.Replace, err error) {
+	var unjoinedReplaces []int
+
+	decoder := json.NewDecoder(bytes.NewReader(out))
+	for {
+		var mod map[string]interface{}
```

Comment:
You can avoid the map-of-string-to-empty-interface with a struct. The Go team defines the shape of the struct and JSON: https://golang.org/ref/mod (search for `type Module struct`). This will save you from having to do type assertion and string based access to the fields.

Reply:
Got it!